docs/deployment-patterns/deployment-as-code.md

```diff
@@ -0,0 +1,113 @@
+
+There is a downside to stopping here: your scripts are managed as code, but your deployment process and variables are still controlled by Octopus. Depending on your situation, this tradeoff might be quite acceptable.
+
+### Getting started
+
+You can follow this process to move your custom scripts without interrupting deployments:
+
+1. Move your deployment scripts from your Octopus deployment process into a file in your application source code
+1. Build your application as normal, but this time with the deployment scripts packaged up alongside your application
+  a. Your deployments should continue to work using the scripts in Octopus
+1. Update your deployment process to use the script from your application package
+
+Now your scripts are colocated with your application source code, all without changing your build pipeline.
+
+Learn about [executing custom scripts in packages](/docs/deployment-examples/custom-scripts/scripts-in-packages/index.md).
+Learn about [custom scripts](/docs/deployment-examples/custom-scripts/index.md).
+
+### Consistency and repeatability
+
+When you move to **scripts as code**, Octopus still makes sure your deployments are consistent and repeatable. Whenever you modify a script, that change flows through the whole process just like the changes to your application code.
+
+### Introducing changes safely {#scripts-as-code-change-safely}
+
+Branches in source control give you the opportunity to test application code changes in isolation before integrating them with other code changes. You can use the exact same approach to introduce changes to your scripts as code. The aim here is to make changes to your scripts without breaking deployments from your `master` branch.
+
+- **Modifying an existing script**: If you modify a script in a branch, that change flows through the whole process just like the changes to your application code. When you deploy a release from that branch, your modified script will be used. When you merge your branch into the `master` branch, your modified script will be used for deployments from the `master` branch.
+- **Adding a new script**: Add an empty script to your `master` branch, configuring Octopus to execute the script, and then author the script content on your branch. This enables you to test your new script in isolation, and merge into `master` when you are ready to integrate.
+- **Deleting a script**: Delete the content of your script in your branch, leaving the empty script file to be packaged. Now you can test your deployment still works with the empty script. Once you are ready, you can configure Octopus to stop calling the script, and delete the script from your `master` branch.
```

Comment:
Not sure this step is necessary to mention?

Reply:
I'd leave it in if this is something users will need to do.

Reply:
I wanted it here for completeness. I'm trying to take an opinion about the process we recommend, as compared to presenting all the possible ways of achieving a thing. I'll leave it in for now.